src/Avalonia.Controls/Window.cs

```diff
@@ -78,9 +78,16 @@ public class Window : WindowBase, IStyleable, IFocusScope, ILayoutRoot, INameSco
         public static readonly StyledProperty<WindowIcon> IconProperty =
             AvaloniaProperty.Register<Window, WindowIcon>(nameof(Icon));
 
+        /// <summary>
+        /// Defines the <see cref="WindowStartupLocation"/> proeprty.
+        /// </summary>
+        public static readonly DirectProperty<Window, WindowStartupLocation> WindowStartupLocationProperty =
+            AvaloniaProperty.RegisterDirect<Window, WindowStartupLocation>(nameof(WindowStartupLocation), o => o.WindowStartupLocation);
```

Comment:
By not providing a setter here you're registering the property as read-only, which is why setting in XAML now doesn't work.